packages/api/tests/task.test.js

```diff
@@ -1862,6 +1879,315 @@ describe('Task tests', () => {
+      const [{ id: soilAmendmentPurposeTwo }] = await mocks.soil_amendment_purposeFactory();
+
+      // Insert three differents products
+      const [{ product_id: soilAmendmentProductOne }] = await mocks.productFactory(
+        { promisedFarm: [{ farm_id }] },
+        // of type 'soil_amendment_task'
+        mocks.fakeProduct({ type: 'soil_amendment_task' }),
+      );
+      const [{ product_id: soilAmendmentProductTwo }] = await mocks.productFactory(
+        { promisedFarm: [{ farm_id }] },
+        // of type 'soil_amendment_task'
+        mocks.fakeProduct({ type: 'soil_amendment_task' }),
+      );
+
+      // Initial task product state
+      const soilAmendmentTaskProductData = [
+        mocks.fakeSoilAmendmentTaskProduct({
+          product_id: soilAmendmentProductOne,
+          purpose_relationships: [
+            { purpose_id: soilAmendmentPurpose },
+            { purpose_id: soilAmendmentPurposeTwo },
+          ],
+        }),
+        mocks.fakeSoilAmendmentTaskProduct({
+          product_id: soilAmendmentProductTwo,
+          purpose_relationships: [{ purpose_id: soilAmendmentPurpose }],
+        }),
+      ];
+
+      const taskData = {
+        ...mocks.fakeTask({
+          soil_amendment_task: mocks.fakeSoilAmendmentTask({
+            method_id: soilAmendmentMethod,
+          }),
+          soil_amendment_task_products: soilAmendmentTaskProductData,
+          task_type_id,
+          owner_user_id: user_id,
+          wage_at_moment: 50,
+          assignee_user_id: user_id,
+        }),
+        locations: [{ location_id }],
+      };
+
+      // Add task
+      postTaskRequest({ user_id, farm_id }, 'soil_amendment_task', taskData, async (err, res) => {
+        expect(res.status).toBe(201);
+        const createdTask = res.body;
+        const createdTaskProducts = createdTask.soil_amendment_task_products;
+        const { task_id } = createdTask;
+        // Delete abandonment reason to prevent validation error
+        delete createdTask.abandonment_reason;
+
+        // Remove all purpose relationships
+        const taskProductForDeletedPurpose = createdTaskProducts.find(
+          (product) => product.product_id == soilAmendmentProductTwo,
+        );
+        const deletedPurposeRelationship = taskProductForDeletedPurpose.purpose_relationships.pop();
+
+        taskProductForDeletedPurpose.purpose_relationships.push(deletedPurposeRelationship);
+        // Delete a task product
+        const deletedTaskProductOne = createdTaskProducts.pop();
+        const deletedTaskProductTwo = createdTaskProducts.pop();
+
+        completeTaskRequest(
+          { user_id, farm_id },
+          {
+            ...createdTask,
+            ...fakeCompletionData,
+          },
+          task_id,
+          'soil_amendment_task',
+          async (err, res) => {
+            expect(res.status).toBe(400);
+
+            // Two active and one deleted task product should be present
```

Comment:
Yeah it can be only one -- I was copying and pasting the other test to be fast with tests, but extra stuff is confusing and slow -- will remove extra.